testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_custom_conflicts.py

```diff
@@ -0,0 +1,1013 @@
+    replicator_authenticator = authenticator.authentication(session_id, cookie, authentication_type="session")
+    repl_config = replicator.configure(cbl_db, sg_blip_url, continuous=False, channels=channels,
+                                       replicator_authenticator=replicator_authenticator,
+                                       replication_type="push_pull")
+    repl = replicator.create(repl_config)
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    replicator.stop(repl)
+    assert total == completed, "total is not equal to completed"
+
+    doc_ids = db.getDocIds(cbl_db)
+    sg_docs = sg_client.get_all_docs(url=sg_url, db=sg_db, auth=session)["rows"]
+
+    # creating conflict for docs on SG
+    sg_client.update_docs(url=sg_url, db=sg_db, docs=sg_docs, number_updates=2,
+                          property_updater=property_updater, auth=session)
+
+    # creating conflict for docs on CBL
+    cbl_docs = db.getDocuments(cbl_db, doc_ids)
+    for doc_id in cbl_docs:
+        for _ in range(2):
+            log_info("Updating CBL Doc - {}".format(doc_id))
+            data = cbl_docs[doc_id]
+            data = property_updater(data)
+            data["cbl_random"] = random_string(length=10, printable=True)
+            db.updateDocument(cbl_db, doc_id=doc_id, data=data)
+
+    repl_config = replicator.configure(cbl_db, sg_blip_url, continuous=True, channels=channels,
+                                       replicator_authenticator=replicator_authenticator,
+                                       replication_type=replicator_type, conflict_resolver="exception_thrown")
+    repl = replicator.create(repl_config)
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    replicator.stop(repl)
+    assert total == completed, "total is not equal to completed"
+
+    # printing doc content before replication conflicted docs
+    try:
+        sg_docs_content = sg_client.get_bulk_docs(sg_url, sg_db, doc_ids, session)[0]
+    except Exception as err:
+        log_info("Requesting for deleted docs from SG: {}".format(err))
+        sg_docs_content = []
+    cbl_docs = db.getDocuments(cbl_db, doc_ids)
+    for sg_doc in sg_docs_content:
+        doc_id = sg_doc["_id"]
+        cbl_doc = cbl_docs[doc_id]
+        assert sg_doc["sg_new_update1"] != cbl_doc["sg_new_update1"], "exception_thrown ccr didn't crash the app"
+        assert sg_doc["sg_new_update2"] != cbl_doc["sg_new_update2"], "exception_thrown ccr didn't crash the app"
+        assert sg_doc["sg_new_update3"] != cbl_doc["sg_new_update3"], "exception_thrown ccr didn't crash the app"
+        assert "random" not in cbl_doc, "exception_thrown ccr didn't crash the app"
+        assert "cbl_random" not in sg_doc, "exception_thrown ccr didn't crash the app"
```

Comment:
Steps from 993 to 1013 are not expected. It should throw an exception during wait_until_replicator_idle . CBL/SGW doc should not get intacted , it should have same changes as before the replication for both SGW and CBL

Reply:
There is lot of repeated code in this file like

Create docs , push/pull replication, update docs on SGW and CBL 
These above steps can have one common method.

Reply:
that's what not equal check mean. They have their previous value and none of them has overridden the other. By intacted, I mean they should retain their changes. May be my wording is not correct. Let me know the way you want to modify the summary.